### ISSUE TITLE
Is it possible to use client with `service_role`?

### ISSUE THREAD

**Issue by lukaszflorczak** [nuxt-community/supabase-module#30]
In one of  my projects I use custom composable for Supabase and now I want to migrate it to `supabase-module`. In the project I have an API request on Nuxt server side which use `service_role` to bypass RLS. I need to update the place where authorized user doesn't have access (it's read-only for them) and the update is based on external request called on server side. 

I wonder if it's possible to use `useSupabaseClient()`, in this case, with `service_role` secret key instead of `public` anon key. Of course only on server side in this specific place.

**Comment by larbish** [comment]
Currently, the client is instantiated only once with the key you add in the `.env` and then [stored in the nuxtApp context](https://github.com/nuxt-community/supabase-module/blob/main/src/runtime/composables/useSupabaseClient.ts#L13). It means that the same client is always used (on both front and server side). It could be possible to add a `serviceRole` boolean param in the `useSupabaseClient` composable in order to use a one time client instantiated with the `service_role` key, itself added via `.env` file but stored in the `private` runtime config. Feel free to make a PR !

**Comment by lukaszflorczak** [comment]
Thanks for the answer and suggestion. I'll try.

**Comment by lukaszflorczak** [comment]
I decided to create new composable for this case. What do you think about? https://github.com/nuxt-community/supabase-module/pull/31

**Comment by larbish** [comment]
I was about to update my response. Indeed it is a better idea to create a new composable.

**Comment by amrnn90** [comment]
Hey guys, has this been resolved? Right now I couldn't get a `service_role` client working and had to use a custom function that's mainly a copy/paste of `serverSupabaseClient`

```ts

export const serverSupabaseServiceRoleClient = (event) => {
  const {
    supabase: { url, client: clientOptions, cookies: cookieOptions },
  } = useRuntimeConfig().public;
  const { supabaseServiceRoleKey } = useRuntimeConfig();

  if (!event.context._supabaseClient) {
    const supabaseClient = createClient(
      url,
      supabaseServiceRoleKey,
      clientOptions
    );
    const token = useCookie(event, `${cookieOptions.name}-access-token`);
    supabaseClient.auth.setAuth(token);
    event.context._supabaseClient = supabaseClient;
    event.context._token = token;
  }
  return event.context._supabaseClient;
};
```

**Comment by larbish** [comment]
Hey @amrnn90, this feature is still not integrated. We were blocked by [the use of client in server routes](https://github.com/nuxt-community/supabase-module/pull/36). But now that it is fixed, we can move on ! @lukaszflorczak starts working on a [PR](https://github.com/nuxt-community/supabase-module/pull/31) but something went wrong with it (merged with 0 commit, don't know what happens). Whatever, we are now able to integrate this new feature so feel free to create a new PR, I'll be happy to help or I'll try to work on it when I have time.

**Comment by larbish** [comment]
[New release](https://github.com/nuxt-community/supabase-module/releases/tag/v0.1.16) available with the service role client ! Please let me know if it is working as expected for you @lukaszflorczak @amrnn90 👍

**Comment by amrnn90** [comment]
Working well for me, thanks @larbish
